[#181] Modify : 짤 업로드 페이지에 TagSlider 컴포넌트 적용

HY219 opened the pull request:
## 📝 작업 내용

- 기존 selectedTag를 보여주던 부분을 TagSlider로 대체
- input 창에 태그 입력 후 reset 해주도록 수정

## 📷 스크린샷 (선택)

## 💬 리뷰 요구사항(선택)

- ~현재 코드에서 TagBadge와 TagSlider에서 tagName뿐만 아니라, tagId도 전달 받아야되는 형태가 되어서, UploadTagslider로 코드가 유사한 파일을 새로 만들어주었습니다.~
  - ~tagId를 넘겨 받지 않도록 코드를 구현할 수도 있을 것 같은데 아직 고민이 부족해서, 필요가 없다면, 공통 컴포넌트를 사용하는 방향으로 추후 리팩토링을 고민해보고 싶습니다.~
 
  **-> tagName만 추출해서, 공통 TagBadge, TagSlider 사용해주었습니다!**
- 기존 TagSearchForm과 UploadTagSearchForm이 겹치는 코드들이 많다 보니 따로 공통 컴포넌트로 작성하여 함께 쓸 수 있는 부분들이 없을지 추후에 고민해보면 좋을 것 같아요.
  - TagSlider 컴포넌트를 적용하는 부분에서도, 만약 tagId를 넘겨받을 필요가 없는 경우, 코드를 중복해서 작성해야 합니다. 예를 들어 CSS 등의 변경사항이 있을 경우 두 파일을 각각 수정해야 하고, 이때 공통으로 변경해야하는 부분을 놓칠 수 있다고 판단했습니다.
  - 나중에 고민해 보면 좋을 것 같아요:)

# 📍 기타 (선택)

- #180 브랜치를 pull 받은 후 작업했습니다.
- 아래 커밋 내용만 봐주시면 됩니다:)
![image](https://github.com/zzalmyu/zzalmyu-frontend/assets/82134668/30725bad-0b78-4c86-9e92-e3543f2c91d4)
- UI 관련 작업은 #180 브랜치에서 작업해서, 따로 봐주시거나 같이 봐주셔도 무방합니다:)


close #181

HY219 committed:
Modify: 기존 selectedTag를 보여주던 부분을 TagSlider로 대체

tagName 만이 아닌 tagId도 같이 넘겨주어 사용하기  UploadTagSlider 생성

HY219 committed:
Modify: input 창에 태그 입력 후 reset

event.currentTarget.reset()

hyeonjinan096 reviewed (commented):
수고하셨습니다:) 
처음 TagSlider에는 tagId와 tagName 받을 수 있도록 했는데 
다른 자동완성 검색창에 $selectedTags가 string[] 형태로 받아와서 tagName만 받도록 수정했습니다. 

공통컴포넌트를 쓰기 위해 upload에 있는 selectedTags에 tagName만 추출해서 전달해주는 것도 방법일 수 있을 것 같긴 한데 
다른 페이지에서는 TagBadge를 사용하고 업로드 페이지에서는 UploadTagBadge를 사용해서 애매하네요
이 부분도 공통화할 때 고민해봐야할 것 같습니다ㅠ

choi-ik reviewed (commented):
<img width="300" alt="image" src="https://github.com/zzalmyu/zzalmyu-frontend/assets/91654577/28ed03f8-eb78-4cf5-a09e-2b0619d3140e">

<img width="620" alt="image" src="https://github.com/zzalmyu/zzalmyu-frontend/assets/91654577/77daa0e7-9329-422e-bf62-b4e8f5a63abb">

첫번째 사진은 UI가 깨지고 있고, 두번째 사진은 태그 슬라이더가 적용이 되어있지 않습니다 ! 확인 부탁드립니다 ㅎㅎ

choi-ik reviewed (commented):
그리고 업로드 페이지를 위해 tagSlider와 searchForm, autoComplete를 새로 만든다는게 조금 걸리긴 하는데, 이 부분은 추후에 기존에 존재하던 공통 컴포넌트를 좀 더 추상화 시켜서 공통적으로 쓸 수 있게 만들어보죠 ㅠ

HY219 commented:
@choi-ik 저희 패드까지만 대응하기로해서 모바일은 고려 안해줬습니다! TagSlider 왜 적용이 안됐죠? 확인해보겠습니다!!
css 수정에 어려움을 겪고 있어서, 우선 대응하기로한 패드까지만 대응하고, 모바일은 추후에 작업해보겠습니다:)

HY219 committed:
Merge branch '#169/style/upload-page-ui' of https://github.com/zzalmyu/zzalmyu-frontend into #181/modify/replace-selectedtag-with-tagslider

HY219 committed:
Refactor: UploadTagSlider 삭제 및 공통 TagBadge, TagSlider 컴포넌트 사용

UploadTagSlider, UploadTagBadge 파악 삭제
selectedUploadTags에서 tagName만 넘기도록해, 공통 TagSlider 사용

HY219 committed:
Style: TagSlider 영역의 width 수정

HY219 committed:
Rename: selectedTag를 selectedUploadTag로 변경

HY219 committed:
Modify: 태그 input 글자 수 제한

1~10 글자 입력 가능
input의 minLength, maxLength 속성 사용

HY219 committed:
Fix: tag 입력시 input 초기화

HY219 committed:
Rename: newTagId, newTagName을 uploadTagId, uploadTagName으로 변경

새 태그, 기존 태그 모두 해당
업로드할 태그에 해당하므로 네이밍 변경

HY219 committed:
Fix: 태그 검색 바에서 enter 시 page 새로고침되는 이슈 해결

Submit 시 기본 동작 막음
event.preventDefault();
이후에 submit 시 동작할 코드 작성

choi-ik reviewed (approved):
LGTM!

hyeonjinan096 reviewed (approved):
LGTMLGTM

HY219 committed:
Merge: #169 PR

HY219 committed:
Merge branch '#169/style/upload-page-ui' of https://github.com/zzalmyu/zzalmyu-frontend into #181/modify/replace-selectedtag-with-tagslider
